data_safe_haven/config/config_class.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from typing import ClassVar, TypeVar
+
+import yaml
+from pydantic import BaseModel, ValidationError
+from yaml import YAMLError
+
+from data_safe_haven.config.context_settings import Context
+from data_safe_haven.exceptions import (
+    DataSafeHavenConfigError,
+    DataSafeHavenParameterError,
+)
+from data_safe_haven.external import AzureApi
+
+T = TypeVar("T", bound="ConfigClass")
+
+
+class ConfigClass(BaseModel, validate_assignment=True):
```

Comment:
I prefer more descriptive class names if possible. How about?

```suggestion
class SerialisableConfig(BaseModel, validate_assignment=True):
```

Reply:
Done in 7a8cc97